spec/predicates/human-review/vcs.md

```diff
@@ -0,0 +1,103 @@
+# Predicate type: VCS Code Reviews
+
+Type URI: (tentative) https://in-toto.io/attestation/human-review/vcs/v0.1
+
+Version: 0.1.0
+
+## Purpose
+
+This attestation type is used to describe code review approvals of the simplest
+kind. The general expectation is that reviews are tied to specific changes in a
+version control system (VCS) such as Git.
+
+## Use Cases
+
+Changes to source code are reviewed by other members of the development team.
+The review may be performed using systems like Gerrit or via reviewing
+mechanisms built into version control forges such as GitHub and GitLab.
+
+## Prerequisite
+
+Understanding of the
+[in-toto attestation specification](https://github.com/in-toto/attestation).
+
+## Model
+
+Typically, software development processes incorporate some type of code review.
+Other members of the development team are expected to study the changes
+proposed and approve them, confirming they meet various expectations pertaining
+to code quality, functionality, and so on. Therefore, the general expectation
+is that this attestation is generated as a confirmation, an approval of some
+specific change, and is performed by developers in the supply chain who have
+code review responsibilities.
+
+## Schema
+
+```json
+{
+    "_type": "https://in-toto.io/Statement/v1",
+    "subject": [
```

Comment:
It would probably be good to discuss what the subject is (that it should point to some code in a VCS) and probably to discuss if the review covers the referenced _change_ or a review of the entire codebase?  Maybe that could be another field in the predicate?

Reply:
That's a great point, I considered supporting a _range_ of commits that comprise a changeset. I'll take a pass at tweaking it.

Reply:
Is it too complex to have the subject point to the tip and include starting commit for the range in the predicate?

Reply:
What if we leaned into ITE-4 to make the subject of a github-specific attestation a pull request itself? I think it could work but we'd then have to identify how we link the commits in the PR to that being used as a source for the build etc.

Reply:
We've been thinking pretty hard about this on our end, but it really comes down to what you're trying to do.

My guess is that people will want to look at a specific commit/merge and ask if it was reviewed.  The attestation should probably provide a link to the review (ITE-4 can probably help there, but maybe it can just be a URI?).  Then I suspect that some users will want to perform that same check for each commit in the log.

If that's true then I think the subject should be the commit where the PR was merged.  IMO we'd only want the attestation for the final commit because any review or approval probably applies to the _sum_ of the changes and not the individual commits in the PR.

That means that for a branch to be considered reviewed you want each commit the branch pointed to to have a review that applies to it, but you don't necessarily care about the intermediate commits since the branch never pointed to them individually and you likely don't want people to use those commits directly anyways.

What goal do the users of this predicate have in mind "how do I know if the all the code in this repo is safe?", "how can I identify which things haven't been reviewed so I can take a closer look at them?", "I'd like to make sure I trust the folks that provided all reviews on this repo so I need to know who they are", "I'd like to make sure the code in this repo was reviewed for problem X".  "I'd like to have confidence that, at some point, the code in this repo was evaluated for problems X, Y, Z, but I don't need to ensure that each individual commit meets that bar".

Reply:
I think I agree. For a v0.1.0, I suggest we leave it as applying to one or more commits, up to the user. In automated systems, I foresee it being issued at the tip of the feature branch, and I think this works from the "was this change reviewed? give me a CR issued for this PR matching the tip of the PR's branch".